### ISSUE TITLE
Azure Synapse Role Assignment Gov Cloud

### ISSUE THREAD

**Issue by jjordonm** [hashicorp/terraform-provider-azurerm#18468]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.9

### AzureRM Provider Version

3.15.0

### Affected Resource(s)/Data Source(s)

azurerm_synapse_role_assignment

### Terraform Configuration Files

```hcl
resource "azurerm_synapse_role_assignment" "synapse_role_assignment" {
  synapse_workspace_id = azurerm_synapse_workspace.synapse.id
  role_name            = "Synapse SQL Administrator"
  principal_id         = azurerm_synapse_workspace.principal_id

  depends_on = [azurerm_synapse_firewall_rule.synapse_firewall_rule]
}
```

### Debug Output/Panic Output

```shell
│ Error: Synapse is not supported in this Azure Environment
│ 
│   with module.synapse.azurerm_synapse_role_assignment.synapse_role_assignment,
│   on modules/synapse/main.tf line 78, in resource "azurerm_synapse_role_assignment" "synapse_role_assignment":
│   78: resource "azurerm_synapse_role_assignment" "synapse_role_assignment" {
```

### Expected Behaviour

Assign Service Principal to synapse sql admin role

### References

(https://github.com/hashicorp/terraform-provider-azurerm/issues/15443)

**Comment by ms-henglu** [comment]
Hi @jjordonm ,

Thank you for taking time to report this issue!

It seems the API is available in US gov env. I've created an issue in the dependency project: https://github.com/manicminer/hamilton/issues/178. Once it's fixed, I'll enable it in azurerm provider.

**Comment by ms-henglu** [comment]
Hi @jjordonm ,

This feature has been supported in https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.27.0, would you please give it a try? Thanks!

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of v3.27.0. As such, I am going to mark this issue as closed.
